File: Backend/BigDataApi/BigDataApi/Models/Song.cs
namespace BigDataApi.Models;

public partial class Song
{
    public string SpotifyId { get; set; } = null!;

    public string Name { get; set; } = null!;

    public bool? IsExplicit { get; set; }

    public int? DurationMs { get; set; }

    public double? Danceability { get; set; }

    public double? Energy { get; set; }

    public int? Key { get; set; }

    public double? Loudness { get; set; }

    public int? Mode { get; set; }

    public double? Speechiness { get; set; }

    public double? Acousticness { get; set; }

    public double? Instrumentalness { get; set; }

    public double? Liveness { get; set; }

    public double? Valence { get; set; }

    public double? Tempo { get; set; }

    public int? TimeSignature { get; set; }

    public int? AlbumId { get; set; }

    public virtual Album? Album { get; set; }

    public virtual ICollection<Record> Records { get; set; } = new List<Record>();

    public virtual ICollection<Artist> Artists { get; set; } = new List<Artist>();
}

File: frontend/src/types/Song.ts
import { Album } from "./Album";
import { Artist } from "./Artist";
import { Record } from "./Record";

export interface Song {
  SpotifyId: string;
  Name: string;
  IsExplicit: boolean;
  DurationMs: number;
  Danceability: number;
  Energy: number;
  Key: number;
  Loudness: number;
  Mode: number;
  Speechiness: number;
  Acousticness: number;
  Instrumentalness: number;
  Liveness: number;
  Valence: number;
  Tempo: number;
  TimeSignature: number;
  AlbumId: number;
  Album: Album | null;
  Records: Record[];
  Artists: Artist[];
}

export interface SongChart {
  [country: string]: number | string;

  SnapshotDate: string;
}
